src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_field.ts

```diff
@@ -0,0 +1,378 @@
+        });
+      }
+      return {
+        isValid: validationErrors.length === 0,
+        errors: validationErrors,
+      };
+    };
+
+    if (isValidationAsync) {
+      return validateAsync({
+        formData,
+        value: valueToValidate,
+        validationTypeToValidate: validationType,
+      }).then(onValidationResult);
+    } else {
+      const validationErrors = validateSync({
+        formData,
+        value: valueToValidate,
+        validationTypeToValidate: validationType,
+      });
+      return onValidationResult(validationErrors);
+    }
+  };
+
+  /**
+   * Handler to change the field value
+   *
+   * @param newValue The new value to assign to the field
+   */
+  const setValue: Field['setValue'] = newValue => {
+    onValueChange();
+
+    const formattedValue = runFormatters(newValue);
+    setStateValue(formattedValue);
+
+    // Update the form data observable
+    form.__updateFormDataAt(path, getOutputValue(formattedValue));
+  };
+
+  /**
+   * Form <input /> "onChange" event handler
+   *
+   * @param event Form input change event
+   */
+  const onChange: Field['onChange'] = event => {
+    const newValue = {}.hasOwnProperty.call(event!.target, 'checked')
+      ? event.target.checked
+      : event.target.value;
+
+    setValue(newValue);
+  };
+
+  /**
+   * As we can have multiple validation types (FIELD, ASYNC, ARRAY_ITEM), this
+   * method allows us to retrieve error messages for certain types of validation.
+   *
+   * For example, if we want to validation error messages to be displayed when the user clicks the "save" button
+   * _but_ in case of an asynchronous validation (for ex. an HTTP request that would validate an index name) we
+   * want to immediately display the error message, we would have 2 types of validation: FIELD & ASYNC
+   *
+   * @param validationType The validation type to return error messages from
+   */
+  const getErrorsMessages: Field['getErrorsMessages'] = (
+    validationType = VALIDATION_TYPES.FIELD
+  ) => {
+    const errorMessages = errors.reduce((messages, error) => {
+      if (
+        error.validationType === validationType ||
+        (validationType === VALIDATION_TYPES.FIELD &&
+          !{}.hasOwnProperty.call(error, 'validationType'))
+      ) {
+        return messages ? `${messages}, ${error.message}` : (error.message as string);
+      }
+      return messages;
+    }, '');
+
+    return errorMessages ? errorMessages : null;
+  };
+
+  const getOutputValue: Field['__getOutputValue'] = (rawValue = value) => serializer(rawValue);
+
+  // -- EFFECTS
+  // ----------------------------------
+  useEffect(() => {
+    if (isPristine) {
+      // Avoid validate on mount
+      return;
+    }
+    form.__validateFields(fieldsToValidateOnChange);
+  }, [value]);
+
+  const field: Field = {
+    path,
+    type,
+    label,
+    helpText,
+    value,
+    errors,
+    form,
+    isPristine,
+    isValidating,
+    isUpdating,
```

Comment:
This is the state that indicates that a user is writing in a field (with the above debounce). Once a user has stopped updating a field, we can display the error below it (if any).

Reply:
I changed it to `isChangingValue` to make it clear _what_ is happening instead of _who_ is doing something and _how_. (that could potentially change in the future).